applications/HDF5Application/custom_io/hdf5_container_component_io.cpp

```diff
@@ -1,384 +1,91 @@
+        } else if constexpr(std::is_same_v<TContainerDataIO, Internals::HistoricalIO> || std::is_same_v<TContainerDataIO, Internals::BossakIO>) {
+            rCommunicator.SynchronizeVariable(rComponent);
+        } else if constexpr(std::is_same_v<TContainerDataIO, Internals::NonHistoricalIO>) {
+            rCommunicator.SynchronizeNonHistoricalVariable(rComponent);
+        } else {
+            static_assert(!std::is_same_v<TContainerDataIO, TContainerDataIO>, "Unsupported TContainerDataIO.");
+        }
     }
 };
+} // namespace NewContainerComponentIOUtilities
 
-void GetContainerItemReferences(std::vector<ElementType*>& rLocalElements,
-                                ElementsContainerType const& rElements)
-{
-    rLocalElements.resize(rElements.size());
-
-#pragma omp parallel for
-    for (int i = 0; i < static_cast<int>(rElements.size()); ++i)
-    {
-        auto it = rElements.begin() + i;
-        rLocalElements[i] = (&(*it));
-    }
-}
-
-void GetContainerItemReferences(std::vector<ConditionType*>& rLocalConditions,
-                                ConditionsContainerType const& rConditions)
-{
-    rLocalConditions.resize(rConditions.size());
-
-#pragma omp parallel for
-    for (int i = 0; i < static_cast<int>(rConditions.size()); ++i)
-    {
-        auto it = rConditions.begin() + i;
-        rLocalConditions[i] = (&(*it));
-    }
-}
-
-void GetContainerItemReferences(std::vector<NodeType*>& rLocalNodes,
-                                NodesContainerType const& rNodes)
-{
-    GetLocalNodes(rNodes, rLocalNodes);
-}
-
-void GetContainerItemReferences(std::vector<Vertex*>& rLocalVertices,
-                                const VertexContainerType& rVertices)
-{
-    rLocalVertices.resize(rVertices.size());
-
-    IndexPartition<std::size_t>(rVertices.size()).for_each([&rVertices,&rLocalVertices](std::size_t i_vertex){
-        // Why is this even allowed in PointerVectorSet?
-        // The const qualifier from rElements is just discarded
-        // without warning when copying to rLocalElements! If this is
-        // intentional, a comment should explain why it is necessary.
-        rLocalVertices[i_vertex] = &*(rVertices.begin() + i_vertex);
-    });
-}
-
-} // unnamed namespace
-
-template <typename TContainerType, typename TContainerItemType, typename... TComponents>
-ContainerComponentIO<TContainerType, TContainerItemType, TComponents...>::ContainerComponentIO(
-    Parameters Settings, File::Pointer pFile, const std::string& rComponentPath)
+template <class TContainerType, class TContainerDataIO, class... TComponents>
+ContainerComponentIO<TContainerType, TContainerDataIO, TComponents...>::ContainerComponentIO(
+    Parameters Settings,
+    File::Pointer pFile,
+    const std::string& rLegacySuffix)
     : mpFile(pFile)
 {
     KRATOS_TRY;
 
     Parameters default_params(R"(
         {
-            "prefix": "",
+            "prefix"           : "",
             "list_of_variables": []
         })");
```

Comment:
Yes and no, It is nice to have it as a method, but then if we rely on the static methods which gives the defaults, then it will be nightmare to find all the defaults. So I would rather have the defaults like this, and in the last user exposed place (which is redundant), so user can see at one glance the whole defaults.